### ISSUE TITLE
Collection member list interface

### ISSUE THREAD

**Issue by SaschaAdler** [crkn-rcdr/Access-Platform#104]
An interface is required for:

- Looking up slugs and adding the resolved objects to a collection's member list
- Removing members from a collection
- Re-ordering the members list, when the collection has `"behavior": "multi-part"`

**Comment by SaschaAdler** [comment]
Much of this work is done in sapindale already, although I don't believe it's been hooked up to the new database. See #84 for the corresponding work in accessor.

**Comment by SaschaAdler** [comment]
Extra checks the lookup that resolves slugs should do:

- [ ] check to make sure the user isn't trying to add the collection to itself
- [ ] check to make sure the user isn't trying to add a member that is already a member of the collection

There should be a `collection.checkAdditions` lapin route that calls a dedicated method in AccessHandler for this; the method should return records that look like

```ts
type AddMemberError =
  | "not-found" // the slug didn't resolve
  | "is-self" // the slug resolved to the collection being edited
  | "already-member" // the slug resolved to an existing member of the collection

type AddMemberRecord = {
  slug: Slug;
  canAdd: true;
  id: Noid;
} | {
  slug: Slug;
  canAdd: false;
  reason: AddMemberError;
}
```

We might want `editCollection` to perform the same checks.

**Comment by DivyaKrishnan26** [comment]
Agreed with the `editCollection` to perform the same check. This looks cool, Thank you @SaschaAdler  :)

**Comment by RussellMcOrmond** [comment]
Marked as "done" in the Kanban, so I bet this should have been closed.
